pymc3/model.py

```diff
@@ -1695,16 +1696,25 @@ def pandas_to_array(data):
     XXX: When `data` is a generator, this will return a Theano tensor!
 
     """
-    if hasattr(data, "values"):  # pandas
```

Comment:
@MarcoGorelli is there any valid pandas input that gets left out by doing `isinstance(data, (pd.Series, pd.DataFrame))` instead of `hasattr(data, "values")`?

Reply:
`pd.Categorical` comes to mind, don't know if you want to support that

Also, it's recommended to use `to_numpy()` instead of `.values` https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.values.html?highlight=values#pandas.Series.values

Reply:
maybe just check if it's an instance of `PandasObject`, which you can import as `from pandas.core.base import PandasObject`?

Reply:
Do they all have the `.isnull()` method?

Reply:
There are other objects which are `PandasObject` which don't have `isnull`, but I don't think you need to worry about people passing them - alternatively, maybe just check if `data` has attribute `to_numpy`, then use `to_numpy` in L1703 so that pd.Categorical and categorical series are supported, otherwise you risk this
```python
>>> s = pd.Series([1,2,3])
>>> s.astype('category')
0    1
1    2
2    3
dtype: category
Categories (3, int64): [1, 2, 3]
>>> s.astype('category').values
[1, 2, 3]
Categories (3, int64): [1, 2, 3]
```
when you really want
```
>>> s.to_numpy()
array([1, 2, 3])
```

Reply:
That's much better, thanks!